### PR TITLE
Documentation: update flatcosi comments, add example.

### PR DESCRIPTION
**Pull request by Gilthoniel**
This adds technical documentation to the code, as well as examples.

### CODE REVIEW

File: cosi/mod.go [hunk 1]
```diff
@@ -13,20 +29,21 @@ type Reactor interface {
 	serde.Factory
```
Comment (nkcr) [inline cosi/mod.go:29]: I would explain why we need a Reactor and how it is used

File: cosi/flatcosi/mod.go [hunk 2]
```diff
@@ -1,3 +1,9 @@
+// Package flatcosi is a flat implementation of a collective signing so that the
+// orchestrator will contact all of the participants to require their signature
```
Comment (nkcr) [inline cosi/flatcosi/mod.go:2]:
```suggestion
// orchestrator will contact all the participants to require their signatures
```

File: cosi/flatcosi/mod.go [hunk 3]
```diff
@@ -93,7 +103,8 @@ type flatActor struct {
 	reactor cosi.Reactor
 }
 
-// Sign implements cosi.Actor. It returns the collective signature of the block.
+// Sign implements cosi.Actor. It returns the collective signature of the
+// message if every participant returns its signature.
```
Comment (nkcr) [inline cosi/flatcosi/mod.go:107]: a threshold, no ?

Comment (Gilthoniel) [reply cosi/flatcosi/mod.go:107]: No this implementation is oriented to be simple, there is no support for a mask and all that stuff that is needed like in the actual threshold package.

File: cosi/mod.go [hunk 4]
```diff
@@ -24,7 +24,9 @@ import (
 	"go.dedis.ch/dela/serde"
 )
 
-// Reactor is an event handler that demultiplex the events.
+// Reactor is a collective signature event handler. Every participant must react
+// to a incoming signature request from the leader, and this abstraction
```
Comment (nkcr) [inline cosi/mod.go:28]:
```suggestion
// to an incoming signature request from the leader, and this abstraction
```

**Review by nkcr — APPROVED** [review]
that would be good if @pierluca can also give his newcomer eye on that.

**Review by pierluca — APPROVED** [review]
It took me a little bit of back-and-forth, but frankly, I don't think this can be any clearer.
LGTM :)

**Pull request merged**